bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/converter/type/NumericTypeConverter.java

```diff
@@ -52,28 +63,120 @@ protected boolean toBindingValidation(Type type) {
     }
 
     @Override
-    protected boolean fromBindingValidation(JsonElement value, String type) {
+    protected boolean fromBindingValidation(JsonElement value, String unit, String type) {
         return true;
     }
 
     @Override
-    protected DecimalType fromBinding(JsonElement value, String type, ChannelType tp, Locale locale)
+    protected State fromBinding(JsonElement value, String unit, String type, ChannelType tp, Locale locale)
             throws ConverterException {
         if ("----".equals(value.getAsString())) {
             return new DecimalType(0);
         } else {
-            return new DecimalType(value.getAsDouble());
+            double dValue = value.getAsDouble();
+
+            String itemType = tp.getItemType();
+
+            if (itemType != null) {
+
+                if (itemType.equals("Number:Temperature")) {
+                    Unit<Temperature> targetUnit = null;
+
+                    if (unit.equals("°C")) {
+                        targetUnit = SIUnits.CELSIUS;
+                    } else if (unit.equals("°C*min")) {
+                        targetUnit = SIUnits.CELSIUS;
+                    } else if (unit.equals("°F")) {
+                        targetUnit = ImperialUnits.FAHRENHEIT;
+                    } else if (unit.equals("°F*min")) {
+                        targetUnit = ImperialUnits.FAHRENHEIT;
+                    }
+
+                    if (targetUnit != null) {
+                        return new QuantityType<>(dValue, targetUnit);
+                    }
+                } else if (itemType.equals("Number:ElectricPotential")) {
+                    Unit<ElectricPotential> targetUnit = null;
+
+                    if (unit.equals("V")) {
+                        targetUnit = Units.VOLT;
+                    }
+
+                    if (targetUnit != null) {
+                        return new QuantityType<>(dValue, targetUnit);
+                    }
+                } else if (itemType.equals("Number:Time")) {
+                    Unit<Time> targetUnit = null;
+
+                    if (unit.equals("s")) {
+                        targetUnit = Units.SECOND;
+                    } else if (unit.equals("m") || unit.equals("min")) {
+                        targetUnit = Units.MINUTE;
+                    } else if (unit.equals("h")) {
+                        targetUnit = Units.HOUR;
+                    } else if (unit.equals("Mois")) {
```

Comment:
This seems like it will only work when gateway language is configured to be French? Related to https://github.com/openhab/openhab-addons/pull/14263#discussion_r1602279155

Reply:
Fix in next commit.
I do all the language of the gateway to have the fullset of possible values.

Reply:
That is really a lot of work I see you have put into supporting many languages. Well done! 👍